automation/check-patch.e2e-kubemacpool-functests.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash -xe
+
+# This script should be able to execute KubeMacPool
+# functional tests against Kubernetes cluster with
+# CNAO built with latest changes, on any
+# environment with basic dependencies listed in
+# check-patch.packages installed and docker running.
+#
+# yum -y install automation/check-patch.packages
+# automation/check-patch.e2e-kubemacpool-functests.sh
+
+teardown() {
+    make cluster-down
+    rm -rf "${TMP_COMPONENT_PATH}"
+}
+
+main() {
+    export COMPONENT="kubemacpool"
+
+    # Spin-up ephemeral cluster with all necessary components
+    source automation/e2e-functests.setup.sh
+
+    cd ${TMP_COMPONENT_PATH}
+
+    # Deploy necessary network addons components with CNAO
+    cat <<EOF > cr.yaml
+apiVersion: networkaddonsoperator.network.kubevirt.io/v1alpha1
+kind: NetworkAddonsConfig
+metadata:
+  name: cluster
+spec:
+  imagePullPolicy: Always
+  kubeMacPool: {}
+  linuxBridge: {}
+  multus: {}
+EOF
+    kubectl apply -f cr.yaml
+    kubectl wait networkaddonsconfig cluster --for condition=Available --timeout=13m
+
+    # Run KubeMacPool functional tests
+     KUBECONFIG=${KUBECONFIG} E2E_TEST_ARGS="--ginkgo.skip='mac-pool is full'" make functest
```

Comment:
can you make sure that `ginkgo.skip` works as expected so that these tests don't run on production?

Reply:
What about running tier1 tests here ? meaning filtering by polarion id.

Reply:
@RamLavi  I passed `-ginkgo.skip=mac-pool is full` it skipped those test

@qinqon I think we should run all tier1 test shouldnt we?

Reply:
> @RamLavi I passed -ginkgo.skip=mac-pool is full it skipped those test

@ormergi why do we skip them? We can specify (limit) the size of the MAC range in the CR.

Reply:
Ram suggested to skip to tests because they might stuck, that comment vanished because I pushed new commit.

When kubemacpool deployed by CNAO, it is managed by CNAO so if we change the macpool range value and restart KMP pods, CNAO will bring KMP pods back with the previous macpool range values.
@RamLavi  is that right?

Reply:
I think we don't control the range by KMP e2e anymore, do we?

Reply:
You right, we dont change the range in e2e anymore 
we do restart KMP pods by manipulating KMP deployment replica number.
I will verify "full macpool" tests are passing, if they are I will remove skip flag.